ddtrace/contrib/pytest/plugin.py

```diff
@@ -162,24 +162,14 @@ def pytest_runtest_makereport(item, call):
 
     try:
         result = outcome.get_result()
-
-        if hasattr(result, "wasxfail") or "xfail" in result.keywords:
-            if result.skipped:
-                # XFail tests that fail are recorded skipped by pytest
-                span.set_tag(test.RESULT, test.Status.XFAIL.value)
-                span.set_tag(test.XFAIL_REASON, result.wasxfail)
-            else:
-                span.set_tag(test.RESULT, test.Status.XPASS.value)
-                if result.passed:
-                    # XPass (strict=False) are recorded passed by pytest
-                    span.set_tag(test.XFAIL_REASON, result.wasxfail)
-                else:
-                    # XPass (strict=True) are recorded failed by pytest, longrepr contains reason
-                    span.set_tag(test.XFAIL_REASON, result.longrepr)
+        xfail = hasattr(result, "wasxfail") or "xfail" in result.keywords
+        skipped = any(x in result.keywords for x in ["skip", "skipif", "skipped"])
```

Comment:
```suggestion
        has_skip_keyword = any(x in result.keywords for x in ["skip", "skipif", "skipped"])
```

?

below seeing this is weird to understand

```python
if result.skipped:
    if xfail and not skipped:
```

Reply:
@brettlangdon That branching as in merging both into one? Or keeping it as before?